backend/cmm_peephole_engine.ml

```diff
@@ -0,0 +1,187 @@
+(* Variable kinds *)
+
+type _ pattern_kind =
+  | Expr : Cmm.expression pattern_kind
+  | Int : int pattern_kind
+  | Natint : Nativeint.t pattern_kind
```

Comment:
Maybe having a `Pattern` submodule (and maybe others also ?) would probably be cleaner for most of the naming around pattern-related things ?

Reply:
ok